docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
+
+![img_11.png](img_11.png) <br>
+**Figure 7: Sequence Diagram showing the logical implementation of executeCommand() for the List Command**
+
+**Mark Command**
+1. The Mark command will mark the appointment input by the user as done. It will set isDone as true and will be used in the Storage class.
+
+**UnMark Command**
+1. The UnMark command will unmark the appointment input by the user as not done or pending. It will se the isDone as false and will be updated accordingly in the Storage class.
+
+### Parser - Class Implementation
+![img_16.png](img_16.png) <br>
+**Figure 8: UML Diagram of Parser Class**
+1. The Parser class takes in a user command and generates a corresponding Command object for veterinary management system tasks such as **add consultation, vaccination or surgery, find, remove, list, and exit**. This implementation makes use of the Command design pattern to encapsulate the behavior of different types of commands, and the parser serves as a factory for creating these commands based on the user input.
+2. The Parser class contains several static methods that parse different types of commands, such as **`ConsultationCommand`**, **`VaccinationCommand`**, **`SurgeryCommand`**, **`FindCommand`**, **`RemoveCommand`**, **`ListCommand`**, and **`ExitCommand`**.
+3. The **`parse()`** method takes a user command as a String named **`task`** and determines the type of command based on the first word of the command. If it matches any of the known command types, it delegates parsing to the corresponding parse method.
+4. If the first word of the command does not match any known type, it throws a **`SniffException`**. Each parse method takes the command String as input and uses substring operations to extract the various parameters of the command. It then creates a new Command object with these parameters and assigns it to the "command" static variable.
+5. Finally, the parse method returns the command object. If an exception occurs during parsing, it throws a **`SniffException`**.
+
+### Storage - Class Implementation 
+![img_18.png](img_18.png) <br>
+**Figure 9: UML Diagram of Storage class**
+1. The Storage class takes in the path of the Sniff storage file.
+2. **`load()`** method loads the contents of the saved file. It calls **`printFileContents(File)`** which will print out all the stored appointments.
+3. **`load()`** method also calls **`addFileContents`** method.
+4. **`addFileContents`** method parses the saved file and identify stored appointments. Depending on the type of appointments, it will then call either **`readConsultationintoAppointmentList()`** / **`readVaccinationintoAppointmentList()`** / **`readSurgeryintoAppointmentList()`** to add these appointment objects into **`ArrayList<Appointment> APPOINTMENTS`**.
+5. If the file is stored in an incorrect format / has missing details, a **`SniffException`** is thrown.
+
+### Find - Find Implementation
+![img_17.png](img_17.png) <br>
```

Comment:
The use of <<create>> twice from SniffTasks to SniffExcepion does feel slightly confusing.
Is this a duplicate?